src/auto/pathGenerator.cpp

```diff
@@ -1,174 +1,121 @@
+//this code was taken from okapilib most credit goes to them
 #include "autolib/auto/pathGenerator.hpp"
+#include <initializer_list>
+#include <vector>
 
 namespace autolib{
-/**
- * PathGenerator::PathGenerator 
- * 
- * @param  {PathGenerator::PathGenerator(} undefined : 
- */
-PathGenerator::PathGenerator(   const std::initializer_list<Point> &iwaypoints,
-                                const okapi::QLength &iminDistanceBetweenPoints){
-    std::vector<Point> tempWaypoints = iwaypoints;
-    if(tempWaypoints.at(0).x.convert(meter) != 0, tempWaypoints.at(0).y.convert(meter) )
-        tempWaypoints.emplace( tempWaypoints.begin(), .00001_mm, .00001_mm );
-    for( Point waypoint: tempWaypoints ){
-        waypoints.path.emplace_back( waypoint.x.convert(meter), waypoint.y.convert(meter) );
+
+PathGenerator::PathGenerator(   const std::initializer_list<double> &iilimits){
+    std::vector<double> ilimits = iilimits;
```

Comment:
Why don't you pass `PathfinderLimits` directly to the function instead of dealing with all this vector stuff?
Even if you don't want to do that, you can simply interact with the `initializer_list` instead of passing it through a vector.
Also `iilimits` is an interesting approach to redundancy =).